### PR TITLE
plugins/languages/neorg: add package neorg

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add package [neorg](https://github.com/nvim-neorg/neorg).

### CODE REVIEW

File: plugins/utils/neorg.nix [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+              level = "warn";
+            };
+            error = {
+              hl = "ErrorMsg";
+              level = "error";
+            };
+            fatal = {
+              hl = "ErrorMsg";
+              level = "5";
+            };
```
Comment (traxys) [inline plugins/utils/neorg.nix:47]: It  seemed  like level == attrName but the last one says nope....

**Review by traxys — APPROVED** [review]
LGTM (if it was tested with at least the mode options as they are complicated)

File: plugins/utils/neorg.nix (hunk 1 above)
Comment (traxys) [reply plugins/utils/neorg.nix:47]: Maybe you could check if the defaults set has a `level` attr and act differently, but I feel it's too complicated

Comment (GaetanLepage) [reply plugins/utils/neorg.nix:47]: Yes, I didn't want to overdo it. Sometimes, a bit more redundant but simpler code is better !

**Pull request merged**